src/Appwrite/Platform/Tasks/schedule.php

```diff
@@ -1,231 +1,221 @@
- *
- */
-$cli
-->task('schedule')
-->desc('Function scheduler task')
-->action(function () use ($register) {
-    Console::title('Scheduler V1');
-    Console::success(APP_NAME . ' Scheduler v1 has started');
-
-    $dbForConsole = getConsoleDB();
+use function Swoole\Coroutine\run;
+
+class Schedule extends Action
+{
+    public const FUNCTION_UPDATE_TIMER = 10; //seconds
+    public const FUNCTION_ENQUEUE_TIMER = 60; //seconds
+
+    public static function getName(): string
+    {
+        return 'schedule';
+    }
+
+    public function __construct()
+    {
+        $this
+            ->desc('Execute functions scheduled in Appwrite')
+            ->inject('dbForConsole')
+            ->inject('getProjectDB')
+            ->callback(fn (Database $dbForConsole, callable $getProjectDB) => $this->action($dbForConsole, $getProjectDB));
+    }
 
     /**
-     * @return  void
-     */
-    $createQueue = function () use (&$functions, &$queue): void {
-        $loadStart = \microtime(true);
+     * 1. Load all documents from 'schedules' collection to create local copy
+     * 2. Create timer that sync all changes from 'schedules' collection to local copy. Only reading changes thanks to 'resourceUpdatedAt' attribute
+     * 3. Create timer that prepares coroutines for soon-to-execute schedules. When it's ready, coroutime sleeps until exact time before sending request to worker.
+    */
+    public function action(Database $dbForConsole, callable $getProjectDB): void
+    {
+        Console::title('Scheduler V1');
+        Console::success(APP_NAME . ' Scheduler v1 has started');
+
         /**
-         * Creating smaller functions list containing 5-min timeframe.
+         * Extract only nessessary attributes to lower memory used.
+         *
+         * @var Document $schedule
+         * @return  array
          */
-        $timeFrame = DateTime::addSeconds(new \DateTime(), FUNCTION_ENQUEUE_TIMEFRAME);
-        foreach ($functions as $function) {
-            $cron = new CronExpression($function['schedule']);
-            $next = DateTime::format($cron->getNextRunDate());
+        $getSchedule = function (Document $schedule) use ($dbForConsole, $getProjectDB): array {
+            $project = $dbForConsole->getDocument('projects', $schedule->getAttribute('projectId'));
+
+            [ $database, $reclaim ] = $getProjectDB($project);
+            $function = $database->getDocument('functions', $schedule->getAttribute('resourceId'));
+            $reclaim();
+
+            return [
+                'resourceId' => $schedule->getAttribute('resourceId'),
+                'schedule' => $schedule->getAttribute('schedule'),
+                'resourceUpdatedAt' => $schedule->getAttribute('resourceUpdatedAt'),
+                'project' => $project,
+                'function' => $function,
+            ];
+        };
+
+        $schedules = []; // Local copy of 'schedules' collection
+        $lastSyncUpdate = DateTime::now();
```

Comment:
Let's move it to L100. 
A bit closer to where this variable is used so we can get a more accurate timestamp